### PR TITLE
feat(MongoNamespace): add MongoNamespace class and use it

### PR DESCRIPTION
**Pull request by daprahamian**

### CODE REVIEW

File: lib/cursor.js [hunk 1]
```diff
@@ -1077,23 +1078,11 @@ Object.defineProperty(Cursor.prototype, 'readPreference', {
...
-      return {
-        database: this.s.ns,
-        collection: ''
-      };
-    }
-    return {
-      database: ns.substr(0, firstDot),
-      collection: ns.substr(firstDot + 1)
-    };
+    console.log(this.s);
```
Comment (kvwalker) [inline lib/cursor.js]: Was this left intentionally?

File: lib/operations/rename.js [hunk 2]
```diff
@@ -25,8 +25,8 @@ class RenameOperation extends OperationBase {
     // Check the collection name
     checkCollectionName(newName);
     // Build the command
-    const renameCollection = `${coll.s.dbName}.${coll.s.name}`;
-    const toCollection = `${coll.s.dbName}.${newName}`;
+    const renameCollection = coll.s.namespace.toString();
```
Comment (kvwalker) [inline lib/operations/rename.js]: Why not use `coll.namespace` here?

File: test/functional/cursor_tests.js [hunk 3]
```diff
@@ -3396,9 +3396,10 @@ describe('Cursor', function() {
         test.equal(null, err);
 
         db.collection('myCollection').find({}, function(error, cursor) {
+          console.log(cursor.s.namespace);
```
Comment (kvwalker) [inline test/functional/cursor_tests.js]: Was this left intentionally?

Comment (mbroadst) [reply test/functional/cursor_tests.js]: removed

File: lib/operations/rename.js (hunk 2 above)
Comment (mbroadst) [reply lib/operations/rename.js]: changed

File: lib/cursor.js (hunk 1 above)
Comment (mbroadst) [reply lib/cursor.js]: removed

**Pull request merged**